src/components/VideoList/VideoList.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import styled from 'styled-components';
+import {
+  VideoListHeader,
+  VideoListHeaderProps,
+} from '../VideoListHeader/VideoListHeader';
+import {
+  VideoButtonCard,
+  VideoButtonCardProps,
+} from '../VideoButtonCard/VideoButtonCard';
+
+/**
+ * Represents the props for the VideoListProps component.
+ *
+ * @interface VideoListProps
+ * @property {VideoListHeaderProps} headerDetails - The video list chapter header details.
+ * @property {Array<VideoButtonCardProps>} allVideos - The array of VideoButtonCardProps representing all the videos.
+ */
+interface VideoListProps {
+  headerDetails: VideoListHeaderProps;
+  allVideos: Array<VideoButtonCardProps>;
+}
+
+/**
+ * Represents the styled component for the sidebar div.
+ */
+const VideoListDiv = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+`;
+
+/**
+ * Renders a list of video button components with a header.
+ *
+ * @param {VideoListProps} headerDetails - The header details for the video list.
+ * @param {Array<VideoButtonCardProps>} allVideos - An array of video details.
+ * @returns {JSX.Element} - The rendered video list component.
+ */
+export const VideoList = ({
+  headerDetails,
+  allVideos,
+}: VideoListProps): JSX.Element => {
+  return (
+    <VideoListDiv>
+      <VideoListHeader
+        surahName={headerDetails.surahName}
+        nameTranslation={headerDetails.nameTranslation}
+        surahNameArabic={headerDetails.surahNameArabic}
+        numberOfVerses={headerDetails.numberOfVerses}
+      />
+      {allVideos.map((video, index) => (
+        <VideoButtonCard
```

Comment:
Can we actually rename video button card and video list to lesson button card and lesson list. All text relevant to that change should also be addressed. i.e. "video 1 of 12" should be "lesson 1 of 12". That I believe is inside the button card itself.